src/main/java/budgettrackerapp/service/category/SimpleCategoryService.java

```diff
@@ -0,0 +1,129 @@
+package budgettrackerapp.service.category;
+
+import budgettrackerapp.dto.CategoryDTO;
+import budgettrackerapp.dto.ExpenditureDTO;
+import budgettrackerapp.dto.UserDTO;
+import budgettrackerapp.entity.Category;
+import budgettrackerapp.entity.User;
+import budgettrackerapp.exeptions.CategoryDoesNotExistException;
+import budgettrackerapp.exeptions.ExpenditureInSpecificCategoryDoesNotExistException;
+import budgettrackerapp.mapper.CategoryMapper;
+import budgettrackerapp.mapper.UserMapper;
+import budgettrackerapp.repository.category.CategoryRepository;
+import budgettrackerapp.repository.user.UserRepository;
+import budgettrackerapp.service.user.UserService;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.time.Month;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional
+@AllArgsConstructor
+public class SimpleCategoryService implements CategoryService {
+
+    private CategoryRepository categoryRepository;
+//    private UserService userService;
+    private CategoryMapper categoryMapper;
+    private UserMapper userMapper;
+    private UserRepository userRepository;
+
+    @Override
+    public CategoryDTO create(CategoryDTO categoryDto, Long userId) {
+        Objects.requireNonNull(categoryDto.getName(), "Please, at least provide name for your category!"); //+++
+        Objects.requireNonNull(userId, "User Id must be provided to create category!");
+        User user = userRepository.findById(userId).get();
+        categoryDto.setUserId(user.getId());
+
+        Category category = categoryMapper.mapToEntity(categoryDto);
+        category.setUser(user);
+        category.setAmount(BigDecimal.ZERO);
```

Comment:
User can't have control above category amount, he can only set amount for expenditure,
        category amount exist only for statistic, so user can see how much money he spent for this category.
        And user can only provide initial sum of money which he wants to spend, he can set it as balance and this balance
        will change depending on user actions, like adding expenditure or deleting it.
        That's how I see logic of this application for myself.
        I can't delete amount from useDto, because I want user to see his amount of money that he spent for a specific category,
        but I can remove setter for amount in userDto, so user won't be able to set amount.